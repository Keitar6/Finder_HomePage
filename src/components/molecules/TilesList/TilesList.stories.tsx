import type { Meta, StoryObj } from '@storybook/react';
import TilesList from './TilesList';

const meta = {
  title: 'Molecules/TilesList',
  component: TilesList,
  tags: ['autodocs'],
  decorators: [(Story) => <>{Story()}</>],
} satisfies Meta<typeof TilesList>;

export default meta;
type Story = StoryObj<typeof meta>;

export const defaultTilesList: Story = {};
