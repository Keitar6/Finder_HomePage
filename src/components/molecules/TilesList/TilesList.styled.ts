import styled from 'styled-components';

export const TilesDescription = styled.div`
  display: flex;
  justify-content: center;
  align-items: baseline;
  flex-direction: column;
  gap: 1rem;
`;

export const TilesContainer = styled.div`
  display: flex;
  justify-content: center;
  align-items: baseline;
  flex-direction: column;

  gap: 1rem;
  padding: 0 20px;
`;
