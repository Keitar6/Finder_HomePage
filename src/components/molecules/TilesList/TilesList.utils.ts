import { TileData } from '../../atoms/Tile/Tile';

export const TILES_ARRAY: TileData[] = [
  {
    name: 'placeholder',
    iconName: 'bell',
  },
  {
    name: 'placeholder1',
    iconName: 'bell',
  },
  {
    name: 'placeholder2',
    iconName: 'bell',
  },
  {
    name: 'placeholder3',
    iconName: 'bell',
  },
  {
    name: 'placeholder4',
    iconName: 'bell',
  },
  {
    name: 'placeholder5',
    iconName: 'bell',
  },
  {
    name: 'placeholder6',
    iconName: 'bell',
  },
  {
    name: 'placeholder7',
    iconName: 'bell',
  },
  {
    name: 'placeholder8',
    iconName: 'bell',
  },
  {
    name: 'placeholder9',
    iconName: 'bell',
  },
  {
    name: 'placeholder10',
    iconName: 'bell',
  },
  {
    name: 'placeholder11',
    iconName: 'bell',
  },
  {
    name: 'placeholder12',
    iconName: 'bell',
  },
  {
    name: 'placeholder13',
    iconName: 'bell',
  },
  {
    name: 'placeholder14',
    iconName: 'bell',
  },
  {
    name: 'placeholder15',
    iconName: 'bell',
  },
  {
    name: 'placeholder16',
    iconName: 'bell',
  },
  {
    name: 'placeholder17',
    iconName: 'bell',
  },
  {
    name: 'placeholder18',
    iconName: 'bell',
  },
  {
    name: 'placeholder19',
    iconName: 'bell',
  },
  {
    name: 'placeholder20',
    iconName: 'bell',
  },
  {
    name: 'placeholder21',
    iconName: 'bell',
  },
  {
    name: 'placeholder22',
    iconName: 'bell',
  },
  {
    name: 'placeholder23',
    iconName: 'bell',
  },
];

export const responsibilityObject = {
  spacing: { xs: 0, sm: 0, md: 0 },
  columns: {
    xs: 2,
    sm: 1,
    md: 1,
  },
  item: {
    xs: 1,
    sm: 1,
    md: 2,
  },
};

export const respoSwitcher = (windowDimensions: {
  width: number | null;
  height: number | null;
}) => {
  if (windowDimensions.width) {
    if (windowDimensions.width > 944) return 6;
    if (windowDimensions.width > 599) return 4;
    else return 6;
  } else return 6;
};
