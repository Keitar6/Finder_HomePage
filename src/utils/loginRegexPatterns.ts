import { InputVariant, RegexPat } from '../components/atoms/Input/Input.types';
import { Getters } from '../types/shared.types';

type LoginRegexProps = Getters<InputVariant, RegexPat>;

const loginRegexPattern: LoginRegexProps = {
  search: {
    pattern: /^[a-zA-Z0-9]+$/,
    required: true,
    maxLength: 36,
    minLength: 8,
  },
};

export default loginRegexPattern;
