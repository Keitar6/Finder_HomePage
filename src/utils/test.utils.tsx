import { PropsWithChildren } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import ThemeProvider from '../theme/ThemeContext';

interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  initialUrl?: string;
}

export function renderWithProviders(
  ui: React.ReactElement,
  {
    initialUrl = '/',
    ...renderOptions
  }: ExtendedRenderOptions = {},
) {
  const Wrapper = ({ children }: PropsWithChildren<object>): JSX.Element => (
    <ThemeProvider>
        {children}
    </ThemeProvider>
  );

  return { ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}

export * from '@testing-library/react';
