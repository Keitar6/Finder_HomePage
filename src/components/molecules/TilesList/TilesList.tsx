import { useWindowDimensions } from '../../../hooks/useWindowDimension';
import Tile, { TileData } from '../../atoms/Tile/Tile';
import Typography from '../../atoms/Typography/Typography';
import * as Styled from './TilesList.styled';
import Grid from '@mui/material/Grid';
import { TILES_ARRAY, respoSwitcher } from './TilesList.utils';
import Icon from '../../atoms/Icon/Icon';
import { useEffect, useState } from 'react';

const TilesList = () => {
  const screenDim = useWindowDimensions();
  let paginationLimiter = respoSwitcher(screenDim);
  const [paginatedArray, setPaginatedArray] = useState<TileData[]>([]);

  const showAllOfTheTiles = () => {
    if (paginationLimiter !== TILES_ARRAY.length) paginationLimiter = TILES_ARRAY.length;
    else paginationLimiter = respoSwitcher(screenDim);
  };

  useEffect(() => {
    setPaginatedArray(TILES_ARRAY.slice(0, paginationLimiter));
  }, [paginationLimiter]);

  return (
    <Styled.TilesContainer>
      <Styled.TilesDescription>
        <Typography variant='header2' color='headerText'>
          Join 2 million+ Australians finding better
        </Typography>
        <Typography variant='largeParagraph' color='textGrey'>
          {
            "Finder's team of 40+ experts will help you find the right choices in over 100 categories"
          }
        </Typography>
      </Styled.TilesDescription>
      <Grid
        container
        justifyContent='flex-start'
        columnGap={2}
        columns={{
          xs: 4,
          sm: 8.8,
          md: 13.5,
        }}
      >
        {paginatedArray.map((tile) => (
          <Grid
            key={tile.name}
            item
            xs={true}
            sm={paginationLimiter === TILES_ARRAY.length ? 2 : true}
            md={paginationLimiter === TILES_ARRAY.length ? 2 : true}
            sx={{ maxWidth: '100%', flexBasis: '48%' }}
          >
            <Tile data={tile} />
          </Grid>
        ))}
        <Styled.PaginationButton onClick={showAllOfTheTiles}>
          <Typography variant='paragraph' color='blueDark'>
            Show more categories
          </Typography>

          <Icon
            size={20}
            name={paginationLimiter !== TILES_ARRAY.length ? 'arrowDown' : 'arrowUp'}
          />
        </Styled.PaginationButton>
      </Grid>
    </Styled.TilesContainer>
  );
};

export default TilesList;
