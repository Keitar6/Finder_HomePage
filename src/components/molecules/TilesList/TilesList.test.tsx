import { renderWithProviders } from '../../../utils/test.utils';
import { cleanup } from '@testing-library/react';
import TilesList from './TilesList';

describe('TilesList', () => {
  const item = () => renderWithProviders(<TilesList />, {});

  afterEach(cleanup);
  it('renders correctly', () => {
    expect(item).toMatchSnapshot();
  });
});
