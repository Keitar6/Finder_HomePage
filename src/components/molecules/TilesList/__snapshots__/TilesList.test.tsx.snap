// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TilesList renders correctly 1`] = `[Function]`;
